include/highfive/bits/H5PropertyList_misc.hpp

```diff
@@ -89,6 +89,31 @@ inline void RawPropertyList<T>::add(const F& funct, const Args&... args) {
 
 // Specific options to be added to Property Lists
 
+#ifdef H5_HAVE_PARALLEL
+
+void MPIOCollectiveMetadata::apply(const hid_t plist) const {
+    auto read = MPIOCollectiveMetadataRead{collective_};
+    auto write = MPIOCollectiveMetadataWrite{collective_};
+
```

Comment:
sigh... there are probably some users out there somewhere that want collective write but independent read, or vice versa.

Reply:
In which case, with property list `plist`, they would ask for collective reading:
```
plist.add(MPIOCollectiveMetadataRead{});
```
or collective writing:
```
plist.add(MPIOCollectiveMetadataWrite{});
```

The third option `MPIOCollectiveMetadata` reduces the typing effort and increases discoverability of the correct settings for the case where all operations are collective.